### PR TITLE
fix: enhancement: onboard user API for onesignal

### PR DESCRIPTION
**Pull request by anantjain45823**
## What are the changes introduced in this PR?

Onboarding new api version for onesignal for User Schema referenced as V2 in code.
[One Signal Docs](https://documentation.onesignal.com/reference/create-user)
## What is the related Linear task?

Resolves INT-2234

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3457/test-report.html

**Review by shrouti1507 — APPROVED** [review]

**Review by shrouti1507 — DISMISSED** [review]

**Review by Gauravudia — APPROVED** [review]

**Comment by utsabc** [comment]
Test cases are failing need to fix it

File: test/integrations/destinations/one_signal/processor/data.ts [hunk 1]
```diff
@@ -1542,4 +1543,5 @@ export const data = [
       },
     },
   },
+  ...data_v2,
```
[the remark this answers is not in the document]
Comment (anantjain45823) [reply test/integrations/destinations/one_signal/processor/data.ts:1546]: Existing tests will be removed after that API is removed

**Comment by anantjain45823** [comment]
For Unit test cases we will cover them afterwards

File: src/v0/destinations/one_signal/utils.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const { getOneSignalAliases } = require('./utils');
+
+// Generated by CodiumAI
```
[the remark this answers is not in the document]
Comment (anantjain45823) [reply src/v0/destinations/one_signal/utils.test.js]: done

**Comment by anantjain45823** [comment]
Moved to https://github.com/rudderlabs/rudder-transformer/pull/3499

**Pull request closed without merging**